op/main.py

```diff
@@ -44,46 +45,50 @@ def _handle_event_link(charm_dir, bound_event):
         # (otherwise, we would never have been called).
         return
 
-    event_hook_path = charm_dir / 'hooks' / bound_event.event_kind.replace('_', '-')
+    if issubclass(bound_event.event_type, op.charm.HookEvent):
+        event_dir = charm_dir / 'hooks'
+    else:
+        raise RuntimeError(f'unable to create a symlink: unsupported event type {bound_event.event_type}')
+    # TODO: Handle function/action events here.
+
+    if not event_dir.exists():
+        raise RuntimeError(f'unable to create event symlink: {event_dir} directory does not exist')
+
+    event_path = event_dir / bound_event.event_kind.replace('_', '-')
     create_link = True
     # Remove incorrect symlinks or files.
-    if event_hook_path.exists():
-        # Non-symlink entries and the ones not pointing to "install"
-        # need to be removed.
-        if not event_hook_path.is_symlink():
-            debugf(f'Hook entry at {event_hook_path} is not a symlink:'
-                   ' attempting to remove it.')
+    if event_path.exists():
+        # Non-symlink entries and the ones not pointing to the charm code file need to be removed.
+        if not event_path.is_symlink():
+            debugf(f'Event entry at {event_path} is not a symlink: attempting to remove it.')
```

Comment:
s/Event entry/Path/

"Event entry" is not something people not reading this code can relate to.